hack/test-cmd.sh

```diff
@@ -74,6 +74,25 @@ function check-curl-proxy-code()
   echo "For address ${full_address}, got ${status} but wanted ${desired}"
   return 1
 }
+
+# TODO: remove this function when we do the retry inside the kubectl commands.
+function kubectl-with-retry()
+{
+  retry=true
+  count=0
+  ERROR_FILE="/tmp/kubectl-error"
+  while $retry && [[ $count -lt 4 ]]; do
```

Comment:
nit: Please add exponential backoff, so that this does not just execute a tight loop. Something as simple as a "sleep(2^count)" should be fine.